indexing-service/src/main/java/org/apache/druid/indexing/compact/CompactionConfigBasedJobTemplate.java

```diff
@@ -70,12 +73,34 @@ public List<CompactionJob> createCompactionJobs(
 
     final List<CompactionJob> jobs = new ArrayList<>();
 
+    CompactionState compactionState = CompactSegments.createCompactionStateFromConfig(config);
+
+    String compactionStateFingerprint = CompactionState.generateCompactionStateFingerprint(
+        compactionState,
+        config.getDataSource()
+    );
+
+    if (segmentIterator.hasNext()) {
+      // If we are going to create compaction jobs for this compaction state, we need to persist the fingerprint -> state
+      // mapping so compacted segments from these jobs can reference a valid compaction state.
+      params.getCompactionStateManager().persistCompactionState(
```

Comment:
The templates should only perform lightweight (i.e. non-IO) read-only operations as `createCompactionJobs` may be called frequently.
We should not do any persistence here.
Instead, the `params` can hold some mapping where we can add this compaction state and perform persistence on-demand (perhaps in the `CompactionJobQueue`).

Reply:
thank you for the guidance. Will work on how to get this out of hot path